javascript-loops/main.js

```diff
@@ -0,0 +1,67 @@
+/* exported getNumbersToTen,
+            getEvenNumbersToTwenty,
+            repeatWord,
+            logEachCharacter,
+            doubleAll,
+            getKeys,
+            getValues
+ */
+function getNumbersToTen() {
+  var numbers = [];
+  var currentNumber = 1;
+  while (currentNumber <= 10) {
+    numbers.push(currentNumber);
+    currentNumber++;
+  }
+  return numbers;
+}
+
+function getEvenNumbersToTwenty() {
+  var evenNumbers = [];
+  var currentNumber = 2;
+  while (currentNumber <= 20) {
+    evenNumbers.push(currentNumber);
+    currentNumber += 2;
+  }
+  return evenNumbers;
+}
+
+function repeatWord(word, times) {
+  var count = 1;
+  var repeated = '';
+  while (count <= times) {
+    repeated += word;
+    count++;
+  }
+  return repeated;
+}
+
+function logEachCharacter(string) {
+  for (var i = 0; i < string.length; i++) {
+    console.log(string[i]);
+  }
+}
+
+function doubleAll(numbers) {
+  var doubled = [];
+  for (var i = 0; i < numbers.length; i++) {
+    doubled.push(numbers[i] *= 2);
```

Comment:
**_Uh oh_**

By using the multiplication assignment operator, this is actually making a permanent change to the original array, which is bad news. Unless the intended purpose of the function is to change the original, you should **_never_** change the contents of an array or object given as an argument. If you do, that is what's called a "side effect", and it leads to many bugs. You should do this without making any changes to the original array

Reply:
I forgot that I made it a multiplication operator before realizing that I should push into the doubled array LOL